### PR TITLE
Analytics

### PR DESCRIPTION
**Pull request by georgecrawford**
Currently implemented:
```
{
	"category": "page",
	"action": "scrolldepth",
	"context": {
		"content": {
			"uuid": "e8813dd4-d00d-11e5-831d-09f7778e7377",
			"title": "Rhythm and booze: the battle to recruit young drinkers"
		},
		"product": "AMP",
		"url": "http://localhost.ft.com:5000/content/amps.html",
		"amp_url": "http://localhost.ft.com:5000/content/e8813dd4-d00d-11e5-831d-09f7778e7377",
		"amp_canonical_url": "http://localhost.ft.com:5000/content/amps.html",
		"referrer": "",
		"scroll_depth": "90"
	},
	"device": {
		"spoor_id": "cikqormsn00003i5mv9zbn5vb"
	},
	"system": {
		"api_key": "7107dae3-7c77-4312-92c6-93a4ba7b79ae",
		"source": "amp-analytics",
		"environment": "d",
		"is_live": false,
		"version": "1.0.0"
	},
	"user": {
		"amp_reader_id": ""
	},
	"time": {
		"amp_timestamp": "1455724643874"
	}
}
```

Questions: 
- how to get URL of clicked link? https://github.com/ampproject/amphtml/issues/2046

### CODE REVIEW

**Comment by georgecrawford** [comment]
Should hopefully send to Spoor once https://github.com/Financial-Times/spoor-api/pull/76 is merged and published.

File: views/article.html [hunk 1]
```diff
@@ -10,8 +10,19 @@
   </style>
   <style>body {opacity: 0}</style><noscript><style>body {opacity: 1}</style></noscript>
   <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <script async custom-element="amp-analytics" src="https://cdn.ampproject.org/v0/amp-analytics-0.1.js"></script>
 </head>
 <body>
+  <amp-analytics config="//localhost:5000/analytics/config.json">
```
Comment (quarterto) [inline views/article.html]: Τhis needs to be a relative URL so it will work in production

File: server/controllers/analytics-config.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+		device: {
+			spoor_id: "${clientId(spoor-id)}"
+		},
+		system: {
+			api_key: "7107dae3-7c77-4312-92c6-93a4ba7b79ae",
+			source: "amp-analytics",
+
+			// TODO: check these
+			environment: (process.env.NODE_ENV === "production" ? "p" : "d"),
+			is_live: (process.env.NODE_ENV === "production"),
```
Comment (quarterto) [inline server/controllers/analytics-config.js]: I prefer `req.app.get('env')` where possible

Comment (georgecrawford) [inline server/controllers/analytics-config.js]: aah, thanks - was wondering how to do that

File: views/article.html (hunk 1 above)
Comment (georgecrawford) [inline views/article.html]: good spot

Comment (georgecrawford) [inline views/article.html]: Fixed

**Comment by georgecrawford** [comment]
NB:

To test cookies, I used `127.0.0.1 localhost.ft.com` in /etc/hosts so the ft.com cookie is sent. However, this breaks the relative URL Matt asked for (`Error: undefined source must start with "https://" or "//" or be relative and served from either https or from localhost. Invalid value: /analytics/config.json`), so when testing that you can use `<amp-analytics config="//localhost:5000/analytics/config.json”>` instead.

**Comment by quarterto** [comment]
Merged in ccb3148

**Pull request closed without merging**